### PR TITLE
Fix bug with saving the default font role (LT-18669)

### PR DESCRIPTION
**Pull request by JamesPrabu**
* Changed the code to save the default font fole
* Added test case to test the code

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/726)

### CODE REVIEW

**Comment by andrew-polk** [comment]
Not knowing the background of this issue, it is difficult to understand what the problem is or what this change does.
Could you write a corresponding test? And perhaps enhance the commit description?

---

Reviewed 1 of 1 files at r1.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/726#-:-LFXvtJ031MqeTtKBZqz:b-gz3wal)*

**Comment by jasonleenaylor** [comment]
Yes @JamesPrabu a unit test would be good. Verify that ldml that has the 'Default' role round trips.

@andrew-polk The Default value in this enum means that the font should have the default attribute set. If it was the only value set then it was not saved because of this buggy condition. Empty is the enum value which means there are no attributes.

@JamesPrabu even though this came up as part of the FieldWorks bug I think a better description for the commit would be:
Fix bug with saving the default font role (LT-18669)

**Comment by andrew-polk** [comment]
Thanks, @jasonleenaylor

**Comment by JamesPrabu** [comment]
I added test case and comment based on andrew-polk's code review comments.

File: SIL.WritingSystems.Tests/LdmlDataMapperTests.cs [hunk 1]
```diff
@@ -788,6 +788,49 @@ public void Roundtrip_LdmlFont()
...
+				};
+				fd2.Urls.Add("http://wirl.scripts.sil.org/font2");
+
+				var wsToLdml = new WritingSystemDefinition("en", "Latn", "", "");
+				wsToLdml.Fonts.Add(fd1);
+				wsToLdml.Fonts.Add(fd2);
+				var ldmlAdaptor = new LdmlDataMapper(new TestWritingSystemFactory());
+				ldmlAdaptor.Write(environment.FilePath("test.ldml"), wsToLdml, null);
+				AssertThatXmlIn.File(environment.FilePath("test.ldml"))
+					.HasAtLeastOneMatchForXpath("/ldml/special/sil:external-resources/sil:font[@name='font1' and @types='default' and @size='2.1' and @minversion='3.1.4' and @features='order=3 children=2 color=red createDate=1996' and @lang='en' and @otlang='abcd' and @subset='unknown']/sil:url[text()='http://wirl.scripts.sil.org/font1']", environment.NamespaceManager);
```
Comment (papeh) [inline SIL.WritingSystems.Tests/LdmlDataMapperTests.cs]: a more rigorous test would be HasSpecifiedNumberOfMatchesForXpath(1

Otherwise, LGTM :+1:

**Comment by papeh** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r3.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/726#-:-LGQFKxzAkCtC3tEpVCC:bnfp4nl)*

**Comment by papeh** [comment]
Reviewed 1 of 1 files at r1.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/libpalaso/726)*

**Pull request merged**